data_validation/schema_validation.py

```diff
@@ -172,3 +216,98 @@ def schema_validation_matching(source_fields, target_fields, exclusion_fields):
                 ]
             )
     return results
+
+
+def parse_allow_list(st):
+    output = {}
+    stack = []
+    key = None
+    for i in range(len(st)):
+        if st[i] == ":":
+            key = "".join(stack)
+            output[key] = None
+            stack = []
+            continue
+        if st[i] == "," and not st[i + 1].isdigit():
+            value = "".join(stack)
+            output[key] = value
+            stack = []
+            i += 1
+            continue
+        stack.append(st[i])
+    value = "".join(stack)
+    output[key] = value
+    stack = []
+    return output
+
+
+def get_datatype_name(st):
+    chars = []
+    for i in range(len(st)):
+        if ord(st[i].lower()) >= 97 and ord(st[i].lower()) <= 122:
+            chars.append(st[i].lower())
+    out = "".join(chars)
+    if out == "":
+        return -1
+    return out
+
+
+# typea data types: int8,int16
+def get_typea_numeric_sustr(st):
+    nums = []
+    for i in range(len(st)):
+        if st[i].isdigit():
+            nums.append(st[i])
+    num = "".join(nums)
+    if num == "":
+        return -1
+    return int(num)
+
+
+# typeb data types: Decimal(10,2)
+def get_typeb_numeric_sustr(st):
+    first_half = st.split(",")[0]
+    second_half = st.split(",")[1]
+    first_half_num = get_typea_numeric_sustr(first_half)
+    second_half_num = get_typea_numeric_sustr(second_half)
+    return first_half_num, second_half_num
+
+
+def string_val(st):
+    return str(st).replace(" ", "")
+
+
+def validate_typeb_vals(source, target):
+    if source[0] > target[0] or source[1] > target[1]:
+        return False, True
+    elif source[0] == target[0] and source[1] == target[1]:
+        return False, False
+    return True, False
+
+
+"""
```

Comment:
Which function does this comment refer to?

Reply:
Fixed the comment.